usr.bin/bsdiff/bsdiff/bsdiff.c

```diff
@@ -147,10 +148,9 @@ int main(int argc,char *argv[])
 		err(1, "%s", argv[1]);
 	}
 
-	if (((old=malloc(oldsize+1))==NULL) ||
-		(lseek(fd,0,SEEK_SET)!=0) ||
-		(read(fd,old,oldsize)!=oldsize) ||
-		(close(fd)==-1)) err(1,"%s",argv[1]);
+	old = mmap(NULL, oldsize+1, PROT_READ, MAP_SHARED, fd, 0);
+	if (old == MAP_FAILED || close(fd) == -1)
+		err(1, "%s", argv[1]);
 
```

Comment:
Here and elsewhere we no longer close the file descriptor.  I think you could just close them after mmap.